modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -402,6 +417,28 @@ protected Experiment internalGetExperiment(Application.Name appName,
     }
 
     /**
+     * Updates the tags for a given Application.
+     *
+     * @param applicationName the Application for which the tags should be added
+     * @param tags            list of tags for the experiment
+     */
+    protected void updateExperimentTags(Application.Name applicationName, Set<String> tags) {
+
+        if (tags == null)
+            tags = Collections.EMPTY_SET;
+
+        // update the tag table, just rewrite the complete entry
+        List<ExperimentTagsByApplication> listAllTags = experimentAccessor.getAllTags(applicationName.toString()).all();
```

Comment:
@AndreaSuckro Why are we retrieving the tags from the `experiment` table? Could we not just call `update()` on  `ExperimentTagAccessor` instead with the newly passed `tags` set in the method parameter? It will also avoid the race condition of tags getting overwritten by 2 different calls at the same time due to the rewrite of the complete entry. Let me know what you think on this.